aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -616,433 +230,229 @@ impl AggregatorData {
+        d.delayed_fields.get(id).expect("Get aggregator failed")
     }
 
     #[test]
     fn test_operations_on_new_aggregator() {
         let resolver = FakeAggregatorView::default();
-        let mut aggregator_data = AggregatorData::default();
-        aggregator_data.create_new_aggregator(aggregator_v1_id_for_test(200), 200);
+        let mut data = DelayedFieldData::default();
+        let id = DelayedFieldID::new(200);
+        let max_value = 200;
 
-        let aggregator = aggregator_data
-            .get_aggregator(aggregator_v1_id_for_test(200), 200)
-            .expect("Get aggregator failed");
+        data.create_new_aggregator(id);
 
-        assert_eq!(aggregator.state, AggregatorState::Create { value: 0 });
-        assert_ok!(aggregator.try_add(100, &resolver));
-        assert_eq!(aggregator.state, AggregatorState::Create { value: 100 });
-        assert!(aggregator.try_sub(50, &resolver).unwrap());
-        assert_eq!(aggregator.state, AggregatorState::Create { value: 50 });
-        assert!(!aggregator.try_sub(70, &resolver).unwrap());
-        assert_eq!(aggregator.state, AggregatorState::Create { value: 50 });
-        assert!(!aggregator.try_add(170, &resolver).unwrap());
-        assert_eq!(aggregator.state, AggregatorState::Create { value: 50 });
-        assert_ok_eq!(
-            aggregator.read_aggregated_aggregator_value(&resolver, ReadPosition::AfterCurrentTxn),
-            50
+        assert_eq!(
+            get_agg(&data, &id),
+            &DelayedChange::<DelayedFieldID>::Create(DelayedFieldValue::Aggregator(0))
+        );
+        assert_ok!(data.try_add_delta(id, max_value, SignedU128::Positive(100), &resolver));
+        assert_eq!(
+            get_agg(&data, &id),
+            &DelayedChange::<DelayedFieldID>::Create(DelayedFieldValue::Aggregator(100))
+        );
+        assert!(data
+            .try_add_delta(id, max_value, SignedU128::Negative(50), &resolver)
+            .unwrap());
+        assert_eq!(
+            get_agg(&data, &id),
+            &DelayedChange::<DelayedFieldID>::Create(DelayedFieldValue::Aggregator(50))
+        );
+        assert!(!data
+            .try_add_delta(id, max_value, SignedU128::Negative(70), &resolver)
+            .unwrap());
+        assert_eq!(
+            get_agg(&data, &id),
+            &DelayedChange::<DelayedFieldID>::Create(DelayedFieldValue::Aggregator(50))
         );
+        assert!(!data
+            .try_add_delta(id, max_value, SignedU128::Positive(170), &resolver)
+            .unwrap());
+        assert_eq!(
+            get_agg(&data, &id),
+            &DelayedChange::Create(DelayedFieldValue::Aggregator(50))
+        );
+        assert_ok_eq!(data.read_aggregator(id, &resolver), 50);
+    }
+
+    fn aggregator_delta_change(delta: i128, max_value: u128) -> DelayedChange<DelayedFieldID> {
+        DelayedChange::Apply(DelayedApplyChange::AggregatorDelta {
+            delta: DeltaWithMax {
+                update: if delta > 0 {
```

Comment:
we have `abs` now?

Reply:
but we don't ever want to do that in production code (i.e. you shouldn't have i128 as you are loosing a bit) - so having it here for tests seems fine?